connectors.go

```diff
@@ -97,74 +86,66 @@ var (
 	ErrUnknownConnector = errors.New("unknown connector")
 )
 
-// New returns a new Connector. The signature is generic to facilitate more flexible caller setup
-// (e.g. constructing a new connector based on parsing a config file, whose exact params
-// aren't known until runtime). However, if you can use the API.New form, it's preferred,
-// since you get type safety and more readable code.
-func New(provider providers.Provider, opts map[string]any) (Connector, error) { //nolint:ireturn
-	switch provider {
-	case providers.Salesforce:
-		return newSalesforce(opts)
-	case providers.Hubspot:
-		return newHubspot(opts)
-	default:
-		return nil, fmt.Errorf("%w: %s", ErrUnknownConnector, provider)
-	}
-}
-
-// newSalesforce returns a new Salesforce Connector, by unwrapping the options and passing them to the Salesforce API.
-func newSalesforce(opts map[string]any) (Connector, error) { //nolint:ireturn
-	var options []salesforce.Option
-
-	c, valid := getParam[common.AuthenticatedHTTPClient](opts, "client")
-	if valid {
-		options = append(options, salesforce.WithAuthenticatedClient(c))
-	}
-
-	w, valid := getParam[string](opts, "workspace")
-	if valid {
-		options = append(options, salesforce.WithWorkspace(w))
-	}
-
-	return Salesforce.New(options...)
+// NewProxyConnector returns a new proxy connector for the given provider with the given build options.
+func NewProxyConnector(
+	provider providers.Provider,
+	opts ...proxy.Option,
+) (*proxy.Connector, error) {
+	return proxy.NewConnector(provider, opts...)
 }
 
-// newHubspot returns a new Hubspot Connector, by unwrapping the options and passing them to the Hubspot API.
-func newHubspot(opts map[string]any) (Connector, error) { //nolint:ireturn
-	var options []hubspot.Option
-
-	c, valid := getParam[common.AuthenticatedHTTPClient](opts, "client")
-	if valid {
-		options = append(options, hubspot.WithAuthenticatedClient(c))
+// NewReadConnector returns a new read connector for the given provider with the given build options.
+func NewReadConnector(
```

Comment:
The use of "proxy.Option" is kind of confusing here. If I'm creating a reader, I'm not expecting to deal with proxy things. I'd say if you're going down this route, just move proxy.Option in to the providers package (`providers.Option`) which would seem more intuitive.